src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,302 @@
+package duke;
+
+import duke.exception.AtEmptyException;
+import duke.exception.ByEmptyException;
+import duke.exception.NumberOutOfBoundsException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Duke {
+    private static final String HORIZONTAL_LINE = "____________________________________________________________";
+    private static final String INDENT = "    ";
+    private static final int START_OF_STRING = 0;
+    public static final String SAD_FACE = "\uD83D\uDE41";
+
+    private static String inputCommand;
+    private static String todoTask;
```

Comment:
Using these string variables as global variables and giving it access to multiple functions is generally bad practise because when something unexpected happens, it is difficult to trace what caused the global variable to take on the unexpected value. You may want to consider limiting the scope of variables to within methods wherever possible.

Reply:
Does this include the constants as well?